hack/ensure-kubectl.sh

```diff
@@ -14,35 +14,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# This has been copied from https://github.com/kubernetes-sigs/cluster-api/blob/v1.6.0/hack/ensure-kubectl.sh
+
 set -o errexit
 set -o nounset
 set -o pipefail
 
+if [[ "${TRACE-0}" == "1" ]]; then
+    set -o xtrace
+fi
+
+
+# shellcheck source=./hack/utils.sh
+source "$(dirname "${BASH_SOURCE[0]}")/utils.sh"
+
 GOPATH_BIN="$(go env GOPATH)/bin/"
-MINIMUM_KUBECTL_VERSION=v1.19.0
-GOARCH="$(go env GOARCH)"
-GOOS="$(go env GOOS)"
+MINIMUM_KUBECTL_VERSION=v1.28.0
+goarch="$(go env GOARCH)"
```

Comment:
Please revert back to GOARCH and GOOS

Reply:
May I ask for the reason? We're doing a bare copy/paste of the source file, if there's a specific reason it would be great having an explicit statement for future changes.

Reply:
I didnt understand this change, and what affect would it make differently than the previous change

Reply:
Technically the change above might be more in line with bash common practices, where userspace variables are generally lowercase.

I'll leave the decision up to @Ankitasw though

Reply:
Sure makes sense to me if this adheres to the best practice 👍